lib/axon/loop.ex

```diff
@@ -458,30 +458,26 @@ defmodule Axon.Loop do
     opts = keyword!(opts, [:steps])
     steps = opts[:steps]
 
-    # TODO: temporarily disabled
-    # while {gradients, model_state, new_state, optimizer_state, gradient_state, gradient_step,
-    # flag = Nx.tensor(1)},
-    # flag do
-    # if Nx.greater_equal(gradient_step, steps - 1) do
     {_, new_model_state, _, new_optimizer_state, new_gradient_state, new_gradient_step, _} =
-      (
-        {updates, new_optimizer_state} =
-          update_optimizer_fn.(gradients, optimizer_state, model_state)
-
-        new_gradient_state = zeros_like(model_state)
-        new_model_state = Axon.Updates.apply_updates(model_state, updates, new_state)
-
-        {gradients, new_model_state, new_state, new_optimizer_state, new_gradient_state, 0,
-         Nx.tensor(0)}
-      )
-
-    #   else
-    #     acc_gradients = deep_merge(gradient_state, gradients, fn x, y -> x + y end)
-
-    #     {gradients, model_state, new_state, optimizer_state, acc_gradients, gradient_step + 1,
-    #      Nx.tensor(0)}
-    #   end
-    # end
+      while {gradients, model_state, new_state, optimizer_state, gradient_state, gradient_step,
+             flag = Nx.tensor(1)},
+            flag do
+        if Nx.greater_equal(gradient_step, steps - 1) do
+          {updates, new_optimizer_state} =
+            update_optimizer_fn.(gradients, optimizer_state, model_state)
+
+          new_gradient_state = zeros_like(model_state)
+          new_model_state = Axon.Updates.apply_updates(model_state, updates, new_state)
+
+          {gradients, new_model_state, new_state, new_optimizer_state, new_gradient_state, 0,
+           Nx.tensor(0)}
+        else
+          acc_gradients = deep_merge(gradient_state, gradients, fn x, y -> x + y end)
+
+          {gradients, model_state, new_state, optimizer_state, acc_gradients, gradient_step + 1,
+           Nx.tensor(0)}
+        end
```

Comment:
Wait, both branches return `Nx.tensor(0)`. Doesn't it mean you ever only do a single pass anyway?

Reply:
Yes, I'm pretty sure this is how I got around larger graphs with if, iirc this reduces memory usage slightly